driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/BatchCursor.java

```diff
@@ -38,6 +39,21 @@ public Publisher<List<T>> next() {
         return Mono.create(sink -> wrapped.next(sinkToCallback(sink)));
     }
 
+    public Publisher<List<T>> nextWithSink(final FluxSink<T> parentSink) {
+        return Mono.create(sink -> wrapped.next(
+                (result, t) -> {
+                    if (!parentSink.isCancelled()) {
+                        if (t != null) {
```

Comment:
Any way to test the first two conditions?

Reply:
Well the test case ensures that sink.error is not called so does test it.

Reply:
It tests the false branch but not the true branch.

Reply:
Normal usage tests that the parent branch is not cancelled. The test will test both, cancelled and not cancelled.  As its a race results may return before the cancellation.

Reply:
Not sure how we can easily test `!parentSink.isCancelled() && t != null`.

Reply:
Added a test - I killed the cursor between next calls :)